### PR TITLE
Document order of multiple solutions

### LINKED ISSUE #1372: Order of solutions
When the termination status is `OPTIMAL`, do the solver need to guarantee that the optimal solution is the first one ?
If this is the case, then this is a bug: https://github.com/jump-dev/MosekTools.jl/issues/66
In fact, we kind of assume it the tests already since we only ever test with the first solution.
Is this in the documentation somewhere ?
I remember we discussed about that, see e.g. https://github.com/jump-dev/JuMP.jl/pull/2512#discussion_r592657852

### PR DESCRIPTION
**Pull request by odow**
Closes #1372

### CODE REVIEW

**Comment by blegat** [comment]
> If multiple results are present, the first result (i.e., `result_index=1`) must be an optimal solution.

This disallows having multiple results in case the termination status is not `OPTIMAL`.
I would rather say something like

> If the termination status is `OPTIMAL`, and multiple results are present, the first result (i.e., `result_index=1`) must be an optimal solution.
> If the termination status is `INFEASIBLE` and one of the solutions is an infeasibility certificate, it should be first result.
> If the termination status is `DUAL_INFEASIBLE` and one of the solutions is a dual infeasibility certificate, it should be the first result.

At least that's what we assume in the tests so it makes sense to make it a requirements.

**Comment by odow** [comment]
How's this now.

File: src/attributes.jl [hunk 1]
```diff
@@ -1000,6 +1004,29 @@ struct RawSolver <: AbstractModelAttribute end
     ResultCount()
 
 A model attribute for the number of results available.
+
+## Order of solutions
+
+A number of attributes contain an index, `result_index`, which is used to refer
+to on of the available results. `result_index` must be an integer between `1`
```
Comment (blegat) [inline src/attributes.jl:1011]: on -> one

Comment (odow) [reply src/attributes.jl]: Fixed.

**Review by blegat — APPROVED** [review]

**Pull request merged**